src/poseidon/septidon/params.rs

```diff
@@ -1,41 +1,62 @@
-use super::super::primitives::{Mds as MdsT, P128Pow5T3Compact, Spec};
-use lazy_static::lazy_static;
+use super::super::primitives::{Mds as MdsT, P128Pow5T3Constants};
 
 /// This implementation can be limited to gate degree 5. However, this mode will not work with
 /// blinding or inactive rows. Enable only with a prover that supports assignments to all n rows.
 pub const GATE_DEGREE_5: bool = false;
 
-/// This implementation supports only the scalar field of BN254 at the moment.
-///
-/// To implement for the Pasta curves, adjust the parameters below, and replace the transition round
-/// by a copy, to get 56 rounds instead of 57.
```

Comment:
This circuit does implement exactly 57 rounds, which is why it is not generic over any field. There should be at least an assert somewhere to check this. Maybe repeat this comment too in case someone wants to implement 56 rounds.